model/extensions/V/vext_arith_insts.sail

```diff
@@ -1379,10 +1379,10 @@ function clause execute(MVVTYPE(funct6, vm, vs2, vs1, vd)) = {
                           let rounding_incr = get_fixed_rounding_incr(result_sub, 1);
                           (result_sub >> 1)['m - 1 .. 0] + zero_extend('m, rounding_incr)
                         },
-        MVV_VMUL     => get_slice_int(SEW, signed(vs2_val[i]) * signed(vs1_val[i]), 0),
-        MVV_VMULH    => get_slice_int(SEW, signed(vs2_val[i]) * signed(vs1_val[i]), SEW),
-        MVV_VMULHU   => get_slice_int(SEW, unsigned(vs2_val[i]) * unsigned(vs1_val[i]), SEW),
-        MVV_VMULHSU  => get_slice_int(SEW, signed(vs2_val[i]) * unsigned(vs1_val[i]), SEW),
+        MVV_VMUL     => mult_to_bits(SEW, true, true, vs2_val[i], vs1_val[i], false),
+        MVV_VMULH    => mult_to_bits(SEW, true, true, vs2_val[i], vs1_val[i], true),
+        MVV_VMULHU   => mult_to_bits(SEW, false, false, vs2_val[i], vs1_val[i], true),
+        MVV_VMULHSU  => mult_to_bits(SEW, true, false, vs2_val[i], vs1_val[i], true),
```

Comment:
All these bools make it a bit hard to follow (it's called "boolean blindness" apparently). I think we should probably add some enums:

```
enum Signedness = { Signed, Unsigned }
enum VectorHalf = { High, Low }

...
MVV_VMULHU   => mult_to_bits(SEW, Unsigned, Unsigned, vs2_val[i], vs1_val[i], High),
```